python/instrumentation/openinference-instrumentation-instructor/src/openinference/instrumentation/instructor/_wrappers.py

```diff
@@ -0,0 +1,231 @@
+                    if resp is not None and hasattr(resp, "dict"):
+                        span.set_attribute(OUTPUT_VALUE, json.dumps(resp.dict()))
+                        span.set_attribute(OUTPUT_MIME_TYPE, "application/json")
+                    return resp
+                except Exception as e:
+                    span.set_status(trace_api.Status(trace_api.StatusCode.ERROR, str(e)))
+                    span.record_exception(e)
+                    raise
+
+        async def async_patched_new_func(*args: Any, **kwargs: Any) -> Any:
+            span_name = "instructor.async_patch"
+            with self._tracer.start_as_current_span(
+                span_name,
+                attributes=dict(
+                    _flatten(
+                        {
+                            OPENINFERENCE_SPAN_KIND: OpenInferenceSpanKindValues.TOOL,
+                            INPUT_VALUE_MIME_TYPE: "application/json",
+                            # TODO(harrison): figure out why i cant use args with _get_input_value
+                            INPUT_VALUE: kwargs,
+                        }
+                    )
+                ),
+                record_exception=False,
+                set_status_on_exception=False,
+            ) as span:
+                try:
+                    resp = await new_func(*args, **kwargs)
+                    if resp is not None and hasattr(resp, "dict"):
+                        span.set_attribute(OUTPUT_VALUE, json.dumps(resp.dict()))
+                        span.set_attribute(OUTPUT_MIME_TYPE, "application/json")
+                    return resp
+                except Exception as e:
+                    span.set_status(trace_api.Status(trace_api.StatusCode.ERROR, str(e)))
+                    span.record_exception(e)
+                    raise
+
+        new_create = async_patched_new_func if func_is_async else patched_new_func
+
+        if client is not None:
+            client.chat.completions.create = new_create
+            return client
+        else:
+            return new_create
+
+
+class _HandleResponseWrapper:
+    def __init__(self, tracer: trace_api.Tracer) -> None:
+        self._tracer = tracer
+
+    def __call__(
+        self,
+        wrapped: Callable[..., Any],
+        instance: Any,
+        args: Tuple[Any, ...],
+        kwargs: Mapping[str, Any],
+    ) -> Any:
+        if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
+            return wrapped(*args, **kwargs)
+
+        if instance:
+            span_name = f"{instance.__class__.__name__}.{wrapped.__name__}"
+        else:
+            span_name = wrapped.__name__
+        with self._tracer.start_as_current_span(
+            span_name,
+            attributes=dict(
+                _flatten(
+                    {
+                        OPENINFERENCE_SPAN_KIND: OpenInferenceSpanKindValues.TOOL,
+                        INPUT_VALUE_MIME_TYPE: "application/json",
```

Comment:
There's also tool definition attributes that probably should be captured

Reply:
ack -- will get this done in follow up